ball_detect/main.c

```diff
@@ -76,25 +84,29 @@ void setup(void) {
 }
 
 void sendc(uint8_t ch) {
-  waitForTX();
-  UDR0 = ch;
+	waitForTX();
+	UDR0 = ch;
 }
 
 ISR(USART_RX_vect) {	
 	PORTD ^= 1 << 3;
-	uint32_t modulo, position;
-	uint8_t read_char = UDR0, i, j;
+	uint8_t read_char = UDR0,  message;
 	if(read_char == my_address) {
-		for(j = 2; j >= 0; j--) {	
-			modulo = vision_result[j];
-			position = 1000000;
-			for(i = 0; i < 7; i++) {
-				sendc((modulo / position) % 10);
-				modulo %= position;
-				position /= 10;
-			}
+		if((address_of_message % 2) == 0) {
+			now = vision_result[address_of_message / 2];
+			message = now & 0xFF;
+			now >>= 8;
+			now &= 0xFF;
+		} else {
+			message = now;
+		}
+		if(address_of_message == 5) {
+			address_of_message = 0;
+		} else {
+			address_of_message++;
```

Comment:
How about replacing this whole `if` with `address_of_message = (address_of_message + 1) % 6`? Did I get it right?